build/dsc/templates/dsc_type.rb.erb

```diff
@@ -4,6 +4,7 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
   require Pathname.new(__FILE__).dirname + '../../' + 'puppet/type/base_dsc'
 
   provide :powershell, :parent => Puppet::Type.type(:base_dsc).provider(:powershell) do
+    confine :powershell_major_version => '5'
```

Comment:
So basically what you are stating is that no user of Windows 10 is going to be able to use this.